GVFS/GVFS.FunctionalTests.LockHolder/AcquireGVFSLock.cs

```diff
@@ -45,6 +41,45 @@ public void Execute()
             }
         }
 
+        private static bool TryGetGVFSEnlistmentRootImplementation(string directory, out string enlistmentRoot, out string errorMessage)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return MacPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
+            }
+            else
+            {
+                enlistmentRoot = null;

```

Comment:
Not able to use WindowsPlatform here - because of its dependency on WindowsIdentity. Let me know if there is a better approach.

Reply:
The only alternative that jumps out to me is moving `TryGetGVFSEnlistmentRootImplementation` and `GetNamedPipeNameImplementation` into a new `WindowsPlatform.Shared.NetStandard.cs` file that has only code that can compile into .NET Standard.

With how limited this is though I think you'd be OK just adding a comment here that describes why the code was copied rather than calling `WindowsPlatform`.  As general rule the functional test code doesn't consume product code anyway, this project is the one exception.

Reply:
Once everything moves to .NET Core for the WindowsIdentity issue we can look at the compatibility pack described here: https://devblogs.microsoft.com/dotnet/announcing-the-windows-compatibility-pack-for-net-core/

Reply:
I agree with William; let's leave a comment here on why we're doing it this way (and that we're intentionally duplicating code). Probably best to leave this comment as a TODO(Mac) so we'll find it whenever we end up moving everything to .NET Core.